### PR TITLE
feat: open weaviate functionalities to docarray

### PR DESCRIPTION
**Pull request by jonathan-rowley**
There are certain features in Weaviate that are not included with DocArray which are helpful to me and may be helpful to others.

Goals:

add the ability to pass additional properties to Weaviate
add the ability to pass sorting options to Weaviate
add the ability to limit return based on Weaviates Certainty score
update the docs

### CODE REVIEW

**Comment by JoanFM** [comment]
Hey @jonathan-rowley,

Thanks for the change, now commit-lint passes, could you add tests for the features you just added?

**Comment by jonathan-rowley** [comment]
@JoanFM any guidance on where and how to do that?

**Comment by JoanFM** [comment]
> @JoanFM any guidance on where and how to do that?

You may inspire yourself by this file:

`tests/unit/array/mixins/test_del.py`

File: docs/advanced/document-store/weaviate.md [hunk 1]
```diff
@@ -290,4 +290,236 @@ Embeddings Nearest Neighbours with "price" at most 7:
...
+    embedding=[3. 3. 3.],    price=3
+    embedding=[4. 4. 4.],    price=4
+    embedding=[5. 5. 5.],    price=5
+    embedding=[6. 6. 6.],    price=6
+    embedding=[7. 7. 7.],    price=7
+    embedding=[8. 8. 8.],    price=8
+    embedding=[9. 9. 9.],    price=9
+```
+
+## Set Minimum Certainty on Query Results
```
Comment (hanxiao) [inline docs/advanced/document-store/weaviate.md:373]:
```suggestion
## Set minimum certainty on query results
```

Comment (hanxiao) [reply docs/advanced/document-store/weaviate.md:373]: any non-h1 title follows first-word captialize case, h1 follows CamelCase

File: docs/advanced/document-store/weaviate.md [hunk 2]
```diff
@@ -290,4 +290,236 @@ Embeddings Nearest Neighbours with "price" at most 7:
...
+
+This should return something similar to:
+
+```bash
+Only results that have a 'weaviate_certainty' of higher than 0.9 should show:
+         text=['Persist Documents with Weaviate.']
+         scores=[{'weaviate_certainty': {'value': 1.0000001}, 'cosine_similarity': {'value': 1.0000002000000001}}]
+```
+
+## Including Additional Properties in the Return
```
Comment (hanxiao) [inline docs/advanced/document-store/weaviate.md:455]:
```suggestion
## Include additional properties in the return
```

**Comment by hanxiao** [comment]
Thanks a lot for your contribution!!! ❤️

**Pull request merged**